ARCore/app/src/main/java/com/tokbox/sample/tokboxar/AugmentedFaceActivity.java

```diff
@@ -0,0 +1,206 @@
+
+public class AugmentedFaceActivity extends AppCompatActivity implements Scene.OnUpdateListener, Session.SessionListener, PublisherKit.PublisherListener {
+  private static final String LOG_TAG = AugmentedFaceActivity.class.getSimpleName();
+  private static final double MIN_OPENGL_VERSION = 3.0;
+
+  // Tokbox variables
+  private static final int RC_VIDEO_APP_PERM = 124;
+  private static String TOKEN  = "YOUR_TOKEN";
+  private static String SESSION_ID = "YOUR_SESSION_ID";
+  private static String API_KEY = "YOUR_API_KEY";
+  private Session mSession;
+  private FrameLayout mPublisherViewContainer;
+
+  // ARCore variables
+  private FaceArFragment mFragment;
+  private ModelRenderable mRenderable;
+  private Texture mTexture;
+  private Scene mScene;
+  private ArSceneView mSceneView;
+
+  private final HashMap<AugmentedFace, AugmentedFaceNode> mFaceNodeMap = new HashMap<>();
+
+  public AugmentedFaceActivity() {
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_augmented_face);
+
+    if(!checkIsSupportedDeviceOrFinish(this)) return;
+
+    // Load the model Renderable from the resource
+    ModelRenderable.builder()
+      .setSource(this, R.raw.fox_face)
+      .build()
+      .thenAccept(modelRenderable -> {
+        mRenderable = modelRenderable;
+        modelRenderable.setShadowCaster(false);
+        modelRenderable.setShadowReceiver(false);
+      });
+
+    // Load the face mesh Texture
+    Texture.builder()
+      .setSource(this, R.drawable.fox_face_mesh_texture)
+      .build()
+      .thenAccept(texture -> mTexture = texture);
+
+    requestPermissions();
+  }
+
+  @AfterPermissionGranted(RC_VIDEO_APP_PERM)
+  private void requestPermissions(){
+    String[] permissions = { Manifest.permission.INTERNET, Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO };
+    if(EasyPermissions.hasPermissions(this, permissions)){
+      // ARCore Initialisation
+      mFragment = (FaceArFragment) getSupportFragmentManager().findFragmentById(R.id.publisher_ar);
+      mSceneView = mFragment.getArSceneView();
+      mSceneView.setCameraStreamRenderPriority(Renderable.RENDER_PRIORITY_FIRST);
+      mScene = mSceneView.getScene();
+      mScene.addOnUpdateListener(this);
+
+      // TokBox Initialisation
+      mPublisherViewContainer = findViewById(R.id.publisher_container);
+      mSession = new Session.Builder(this, API_KEY, SESSION_ID).build();
+      mSession.setSessionListener(this);
+      mSession.connect(TOKEN);
+    }else{
+      EasyPermissions.requestPermissions(this, "This app needs access to your camera and mic to make a video calls", RC_VIDEO_APP_PERM, permissions);
+    }
+  }
+
+  // SessionListener methods
+  @Override
+  public void onConnected(Session session) {
+    Log.i(LOG_TAG, "Session Connected");
+    CustomVideoCapturer mCapturer = new CustomVideoCapturer(mFragment.getArSceneView());
```

Comment:
Not very important comment :) but
You usually prefix the variable names with an `m` if they are instance fields, but mCapturer is more a local variable rather than a instance field.

https://en.wikipedia.org/wiki/Hungarian_notation